common/credentials.go

```diff
@@ -0,0 +1,79 @@
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/spf13/viper"
+)
+
+type AwsCredentials struct {
+	ClientRegion   string
+	AwsAccessKeyId string
+	AwsSecretKey   string
+	AssumeRoleArn  string
+}
+
+func WithAwsCredentials() AwsCredentials {
+	return AwsCredentials{
+		ClientRegion:   viper.GetString("aws.clientRegion"),
+		AwsAccessKeyId: viper.GetString("aws.staticCredentials.awsAccessKeyId"),
+		AwsSecretKey:   viper.GetString("aws.staticCredentials.awsSecretKey"),
+		AssumeRoleArn:  viper.GetString("aws.assumeRoleArn"),
+	}
+}
+
+// assume role takes precedence over statc credentials if both set
```

Comment:
why take precedence? seems it's more logical to use the static credentials to perform assume role, no?

Reply:
@realstraw I was thinking more on the hyperforce scenario, but let me change to be consistent with the aws credential chain precedence order.

Reply:
@realstraw changed

Reply:
what I'm thinking is if you have both static and assume role, you use the static cred to get the sts client and assume the role. If you have static without assume role, then just static. If you have just assume role, you assume role with default chain

Reply:
@realstraw ok, updated to logic above